### DISCUSSION TITLE
Probleme mit der Ausrichtung wg. Drehung des Bildes nach Update auf 12.0.1

### DISCUSSION THREAD

**Discussion by ManfMert** [jomjol/AI-on-the-edge-device#1408] [Q&A] [answered]
Guten Morgen,

nach dem Update auf die aktuelle Version 12.0.1 sah zunächst alles gut aus, irgendwie klappt aber bei den regelmäßigen Aufnahme die Ausrichtung nicht richtig. Wg. der Reflektionen muss ich die Kamera um ca. 90 drehen und dann das Bild entsprechend zurückdrehen.

Nach der Erstellung des Reference-Image sieht die Ausrichtung zunächst gut aus.
![Reference Image](https://user-images.githubusercontent.com/24528404/203916870-bac826b6-566d-44e5-a375-f15b37169588.jpg)
![Digital ROIs](https://user-images.githubusercontent.com/24528404/203916444-da6cb8be-0a16-46da-8f39-f2c5a4dba3e8.jpg)
![Analog ROIs](https://user-images.githubusercontent.com/24528404/203916508-e8cb593c-c681-4825-ab02-3677d41c5e6b.jpg)

Auch nach dem Reboot ist die Welt noch in Ordnung.
![Overview after Reboot](https://user-images.githubusercontent.com/24528404/203916767-4fd79fb4-f841-49c9-a085-e96419e02553.jpg)

Alle weiteren Aufnahmen sind dann aber aus der Richtung und die Auswertung für die Tonne.
![Overview after Reboot   2nd picture](https://user-images.githubusercontent.com/24528404/203916893-016ceb5e-1200-4018-8e6b-5756ac93cc98.jpg)

Ich habe die Schleife Reference-Image > Save & Reboot > Ausrichtung Digital & Analog > Save & Reboot > Test ... mehrmals durchlaufen und komme immer zu dem gleichen Eregbnis!?

Ich habe auch versucht, ohne Drehung auszukommen und die Reflektionen mit Klebeband, Umlenkspiegel, Milchglas einer defekten LED, ... abzuschwächen. Die LED ist aber auf der kurzen Distanz zu stark und die Reflektionen nicht in den Griff zu kriegen.

Viele Grüße
Manfred

**Comment by forrestfunk** [comment, marked as answer]
Kann leider nicht helfen, aber ich habe das gleiche Problem. Musste auch aus _Gründen_ praktisch um ~90° gedreht einbauen und drehe nun im reference Image um 95° zurück. Wie beim OP passt es in der Recognition oder Overview aber nicht mehr. 

Kann das dran liegen, dass die "alignment marks" vielleicht schlecht gewählt/nicht gut erkannt werden? Bei mir funktioniert die recognition glücklicherweise trotzdem...

€: interessanterweise war das alignment der Zähler jetzt gerade einmal perfekt, wie gewollt, nach einem Neustart war das Bild aber wieder "falsch". Gefühlt ist das bei jedem reboot wieder bisschen anders.

€2: also bei mir lag das Problem wirklich an den alignment marks. Ich habe mir nun so kleine marker (wie bei den crash-test dummies, schwarz weißer Kreis mit Kreuz) gezeichnet und auf den Zähler geklebt. Jetzt funktioniert die Sache eigentlich einwandfrei . Also Manfred vielleicht versuchst du auch mal bisschen mit deinen alignment marks zu spielen.

Grüße

**Comment by ManfMert** [comment]
Vielen Dank für den Vorschlag.

Ich habe das mit den besseren Referenzpunkten auch mal umgesetzt und das Ergebnis sieht jetzt deutlich besser aus, obwohl die Erkennung noch etwas Training benötigt.

![After Reboot mit neuen Referenzpunkten](https://user-images.githubusercontent.com/24528404/204085883-814fe6bf-a48b-45e6-aa69-5ec4fb846496.jpg)

**Reply by jomjol** [reply]
Du könntest für die Digits noch auf eines der "moderneren" neuronal Netze umstellen (z.B. `dig-cont_0600_s3.tflite`, ggf. noch hochladen). Dann verschwindet das "N", da Zwischenwerte erkannt werden.
